hal/armv7m/imxrt/_init.S

```diff
@@ -179,10 +179,20 @@ _exceptions_dispatch:
 	cpsid if
 
 	mrs r0, psp
+	mov r2, sp
+	mov r3, r0
 	stmdb sp!, {r0, r4-r11, lr}
 
 	mrs r0, ipsr
 	mov r1, sp
+
+	cmp lr, RET_THREAD_PSP
+	bne 1f
+	cmp r3, NULL
+	beq 1f
+	mov r2, r3
+1:
+	push {r2}
```

Comment:
Will the stack be unaligned to 8?

Reply:
It will, thanks for noticing, I've padded the stack to 8.